### DISCUSSION TITLE
need help -> new to ical4j; first tries

### DISCUSSION THREAD

**Discussion by GuKKDevel** [ical4j/ical4j#559] [General]
### Discussed in https://github.com/ical4j/ical4j/discussions/542

<div type='discussions-op-text'>

<sup>Originally posted by **GuKKDevel** January 12, 2022</sup>
Hello, 
I am trying to use ical4j in an Eclipse-IDE. So I downloaded from Sourceforge the latest version 3.0.0 of ical4j, added slf4j and slf4j-simple from the maven repository, same with commons-lang3 and commons-collections4 and cache-api-1.1.1 from the nexus. Included the jars into my classpath.

I coded your example

> FileInputStream fin = new FileInputStream(inPfad + kalender[0] + ".ics");
> CalendarBuilder builder = new CalendarBuilder();
> Calendar calendar = builder.build(fin);`

Now I get 

>[main] INFO net.fortuna.ical4j.util.Configurator - ical4j.properties not found.
>Exception in thread "main" javax.cache.CacheException: No CachingProviders have been configured
>	at javax.cache.Caching$CachingProviderRegistry.getCachingProvider(Caching.java:391)
>	at javax.cache.Caching$CachingProviderRegistry.getCachingProvider(Caching.java:361)
>	at javax.cache.Caching.getCachingProvider(Caching.java:151)
>	at net.fortuna.ical4j.util.JCacheTimeZoneCache.<init>(JCacheTimeZoneCache.java:15)
>	at java.base/java.util.Optional.orElseGet(Optional.java:369)
>	at net.fortuna.ical4j.model.TimeZoneLoader.cacheInit(TimeZoneLoader.java:275)
>	at net.fortuna.ical4j.model.TimeZoneLoader.<init>(TimeZoneLoader.java:81)
>	at net.fortuna.ical4j.model.TimeZoneRegistryImpl.<init>(TimeZoneRegistryImpl.java:125)
>	at net.fortuna.ical4j.model.TimeZoneRegistryImpl.<init>(TimeZoneRegistryImpl.java:116)
>	at net.fortuna.ical4j.model.DefaultTimeZoneRegistryFactory.createRegistry(DefaultTimeZoneRegistryFactory.java:48)
>	at net.fortuna.ical4j.data.CalendarBuilder.<init>(CalendarBuilder.java:105)
>	at Ausprobieren.<init>(Ausprobieren.java:41)
>	at Ausprobieren.main(Ausprobieren.java:66)

Don't know where the mistake is. can you help me?

Kind Regards
GuKKDevel (Karl-Heinz)</div>

**Comment by benfortuna** [comment]
As a workaround, try creating an empty file called "ical4j.properties" in your root source directory. We'll try to improve the experience in future.

**Comment by GuKKDevel** [comment]
As suggested, created the file but still get

>Wed Jan 19 11:27:56 CET 2022
>Exception in thread "main" javax.cache.CacheException: No CachingProviders have been configured
>	at javax.cache.Caching$CachingProviderRegistry.getCachingProvider(Caching.java:391)
>	at javax.cache.Caching$CachingProviderRegistry.getCachingProvider(Caching.java:361)
>	at javax.cache.Caching.getCachingProvider(Caching.java:151)
>	at net.fortuna.ical4j.util.JCacheTimeZoneCache.<init>(JCacheTimeZoneCache.java:15)
>	at java.base/java.util.Optional.orElseGet(Optional.java:369)
>	at net.fortuna.ical4j.model.TimeZoneLoader.cacheInit(TimeZoneLoader.java:275)
>	at net.fortuna.ical4j.model.TimeZoneLoader.<init>(TimeZoneLoader.java:81)
>	at net.fortuna.ical4j.model.TimeZoneRegistryImpl.<init>(TimeZoneRegistryImpl.java:125)
>	at net.fortuna.ical4j.model.TimeZoneRegistryImpl.<init>(TimeZoneRegistryImpl.java:116)
>	at net.fortuna.ical4j.model.DefaultTimeZoneRegistryFactory.createRegistry(DefaultTimeZoneRegistryFactory.java:48)
>	at net.fortuna.ical4j.data.CalendarBuilder.<init>(CalendarBuilder.java:105)
>	at Ausprobieren.<init>(Ausprobieren.java:41)
>	at Ausprobieren.main(Ausprobieren.java:66)

**Comment by benfortuna** [comment]
Yup, so add this line in ical4j.properties:

`net.fortuna.ical4j.timezone.cache.impl=net.fortuna.ical4j.util.MapTimeZoneCache`
